app/controllers/api/v2/job_invocations_controller.rb

```diff
@@ -195,9 +195,15 @@ def job_invocation_params
         end
         job_invocation_params[:inputs] ||= {}
         job_invocation_params[:inputs].permit!
+        permit_provider_inputs job_invocation_params
         @job_invocation_params = job_invocation_params
       end
 
+      def permit_provider_inputs(invocation_params)
+        providers = RemoteExecutionProvider.providers.values.reject { |provider| !provider.respond_to?(:provider_input_namespace) || provider.provider_input_namespace.empty? }
+        providers.map { |provider| invocation_params.dig(*provider.provider_input_namespace)&.permit! }
```

Comment:
`permit!` modifies the parameters in-place?

Reply:
Yes, inspired by what [job_invocation_params](https://github.com/theforeman/foreman_remote_execution/blob/master/app/controllers/api/v2/job_invocations_controller.rb#L197) does. Do you have a different approach in mind?

Reply:
If we just do this for the side effect then we could `each` instead of `map` since we discard the result anway

Reply:
Replaced with `each`.